ddl/ddl_api.go

```diff
@@ -2002,6 +2012,9 @@ func setDefaultAndComment(ctx sessionctx.Context, col *table.Column, options []*
 			col.Flag |= mysql.OnUpdateNowFlag
 			setOnUpdateNow = true
 		case ast.ColumnOptionGenerated:
+			if err := checkIllegalFn4GeneratedColumn(col.Name.L, opt.Expr); err != nil {
```

Comment:
Could we put this check to `checkModifyGeneratedColumn`? 
And I think it's best to unify the check for the generated column, otherwise, it's easy to forget. However, this may need to be handled in another PR

Reply:
Yes, it's better to put the checks for generated columns together. There's several other problems for generated columns, maybe we should make another PR to refactor this place later.

Reply:
In this PR, Could we put this check to `checkModifyGeneratedColumn`?

Reply:
@zimulala  I considered to put this check into `checkModifyGeneratedColumn` before, but changed my mind later because that means we need to go through the `Options` one more time. 
What do you think about this?~

Reply:
It seems that the `ColumnOptionGenerated` option has only one per column and we only do `checkIllegalFn4GeneratedColumn` when the `ColumnOptionGenerated` option is selected. So we needn't
 to traverse the `options` in `checkModifyGeneratedColumn`?

Reply:
OK, I'll put `checkIllegalFn4GeneratedColumn` into `checkModifyGeneratedColumn` here